packages/frontend/src/pages/Users/components/UsersTable.tsx

```diff
@@ -1,121 +1,159 @@
-  const [selectedQuantifier, setSelectedQuantifier] = React.useState<UserDto>();
-
-  const columns = React.useMemo(
-    () => [
-      {
-        Header: 'Id',
-        accessor: '_id',
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        Cell: (data: any): JSX.Element => {
-          return (
-            <div className="flex items-center w-full">
-              <div className="flex items-center">
-                <UserAvatar user={data.row.original} />
-              </div>
-              <div className="flex-grow p-3 whitespace-nowrap">
-                {data.row.original.accounts?.length > 0 ? (
-                  <div>{getUsername(data.row.original)}</div>
-                ) : (
-                  ''
-                )}
-                {shortenEthAddress(data.row.original.ethereumAddress)}
-              </div>
-            </div>
-          );
-        },
-      },
-    ],
-    []
+  const allUsers = useRecoilValue(AllUsers);
+  const [tableData, setTableData] = React.useState<UserDto[]>();
+  const [selectedRole, setSelectedRole] = React.useState<UserRole>(
+    UserRole.USER
   );
+  const [filter, setFilter] = React.useState<string>('');
+  const [filterType, setFilterType] = React.useState<string>('user');
+  const [page, setPage] = React.useState<number>(1);
+  const [lastPage, setLastPage] = React.useState<number>(0);
 
-  const options = {
-    columns,
-    data: allQuantifierUsers ? allQuantifierUsers : [],
-  } as TableOptions<{}>;
-  const tableInstance = useTable(options);
+  React.useEffect(() => {
+    if (allUsers) {
+      setTableData(allUsers);
+    }
+  }, [allUsers]);
 
-  const { getTableProps, getTableBodyProps, rows, prepareRow } = tableInstance;
+  React.useEffect(() => {
+    if (filter && tableData) {
+      let filteredData: UserDto[];
+      if (filterType === 'user') {
+        filteredData = tableData.filter((user) => {
+          const username = getUsername(user)?.toLowerCase();
+          if (username?.includes(filter.toLocaleLowerCase())) {
+            return user;
+          }
+        });
+      } else {
+        filteredData = tableData.filter((user) => {
+          if (
+            user.ethereumAddress?.toLocaleLowerCase() ===
+            filter.toLocaleLowerCase()
+          ) {
+            return user;
+          }
+        });
+      }
+      setTableData(filteredData);
+    }
+  }, [filterType, filter]);
 
-  if (!allQuantifierUsers)
-    return <div>There are no users in the Quantifier pool.</div>;
+  React.useEffect(() => {
+    switch (selectedRole) {
+      case UserRole.USER:
+        setTableData(allUsers);
+        break;
+      case UserRole.ADMIN:
+        setTableData(allAdminUsers);
+        break;
+      case UserRole.FORWARDER:
+        setTableData(allForwarderUsers);
+        break;
+      case UserRole.QUANTIFIER:
+        setTableData(allQuantifierUsers);
+        break;
+    }
+    setFilter('');
+  }, [selectedRole]);
```

Comment:
getting a lint warning: `React Hook React.useEffect has missing dependencies: 'allAdminUsers', 'allForwarderUsers', 'allQuantifierUsers', and 'allUsers'. Either include them or remove the dependency array`